File: modules/Plugin/src/daybreak/abilitywar/utils/base/minecraft/entity/health/Healths.java
package daybreak.abilitywar.utils.base.minecraft.entity.health;

import daybreak.abilitywar.utils.base.minecraft.entity.health.event.PlayerSetHealthEvent;
import org.bukkit.Bukkit;
import org.bukkit.attribute.Attribute;
import org.bukkit.entity.Player;

public class Healths {

	private Healths() {}

	public static double setHealth(final Player player, final double health) {
		final PlayerSetHealthEvent event = new PlayerSetHealthEvent(player, health);
		Bukkit.getPluginManager().callEvent(event);
		if (!event.isCancelled()) {
			player.setHealth(Math.min(event.getHealth(), player.getAttribute(Attribute.GENERIC_MAX_HEALTH).getValue()));
		}
		return player.getHealth();
	}

}

File: modules/Plugin/src/daybreak/abilitywar/utils/base/minecraft/entity/health/Healths.kt
package daybreak.abilitywar.utils.base.minecraft.entity.health

import daybreak.abilitywar.utils.base.minecraft.entity.health.event.PlayerSetHealthEvent
import org.bukkit.Bukkit
import org.bukkit.attribute.Attribute.GENERIC_MAX_HEALTH
import org.bukkit.entity.Player
import kotlin.math.max
import kotlin.math.min

object Healths {

	@JvmStatic
	fun setHealth(player: Player, health: Double): Double {
		val event = PlayerSetHealthEvent(player, max(min(health, player.getAttribute(GENERIC_MAX_HEALTH)!!.value), 0.0))
		Bukkit.getPluginManager().callEvent(event)
		if (!event.isCancelled) {
			player.health = min(event.health, player.getAttribute(GENERIC_MAX_HEALTH)!!.value)
		}
		return player.health
	}

}
